xs/src/libslic3r/Polyline.cpp

```diff
@@ -251,4 +252,105 @@ ThickPolyline::reverse()
+                pp.erase(pp.begin() + id_candidate_first_point);
+                changes = true;
+                polyline->endpoints.first = false;
+                polyline->endpoints.second = false;
+            } else {
+
+                if (nbCandidate_first_point == 1) {
+                    if (polyline->first_point().coincides_with(pp[id_candidate_first_point].first_point())) pp[id_candidate_first_point].reverse();
+                    //concat at front
+                    polyline->width[0] = std::max(polyline->width.front(), pp[id_candidate_first_point].width.back());
+                    polyline->points.insert(polyline->points.begin(), pp[id_candidate_first_point].points.begin(), pp[id_candidate_first_point].points.end() - 1);
+                    polyline->width.insert(polyline->width.begin(), pp[id_candidate_first_point].width.begin(), pp[id_candidate_first_point].width.end() - 1);
+                    polyline->endpoints.first = pp[id_candidate_first_point].endpoints.first;
+                    pp.erase(pp.begin() + id_candidate_first_point);
+                    changes = true;
+                    if (id_candidate_first_point < i) {
+                        i--;
+                        polyline = &pp[i];
+                    }
+                    if (id_candidate_last_point > id_candidate_first_point) {
+                        id_candidate_last_point--;
+                    }
+                } else if (nbCandidate_first_point == 0) {
+                    //update endpoint
+                    polyline->endpoints.first = true;
+                }
+                if (nbCandidate_last_point == 1) {
+                    if (polyline->last_point().coincides_with(pp[id_candidate_last_point].last_point())) pp[id_candidate_last_point].reverse();
+                    //concat at back
+                    polyline->width[polyline->width.size() - 1] = std::max(polyline->width.back(), pp[id_candidate_last_point].width.front());
+                    polyline->points.insert(polyline->points.end(), pp[id_candidate_last_point].points.begin() + 1, pp[id_candidate_last_point].points.end());
+                    polyline->width.insert(polyline->width.end(), pp[id_candidate_last_point].width.begin() + 1, pp[id_candidate_last_point].width.end());
+                    polyline->endpoints.second = pp[id_candidate_last_point].endpoints.second;
+                    pp.erase(pp.begin() + id_candidate_last_point);
+                    changes = true;
+                    if (id_candidate_last_point < i) {
+                        i--;
+                        polyline = &pp[i];
+                    }
+                } else if (nbCandidate_last_point == 0) {
+                    //update endpoint
+                    polyline->endpoints.second = true;
+                }
+
+                if (polyline->last_point().coincides_with(polyline->first_point())) {
```

Comment:
this should never happen, loop creation is handled separately

Reply:
You should be right.